### PR TITLE
Dolci/exchange fwi demo test

### PR DESCRIPTION
**Pull request by Ig-dolci**
# Description

Add `FIREDRAKE_CI` environment in test_demos.

### CODE REVIEW

**Review by dham — CHANGES_REQUESTED** [review]
This deals with the proximate problem but violates single source of truth. We need to put a mode into the demo that makes it faster (possibly by branching on an environment variable).

File: demos/full_waveform_inversion/full_waveform_inversion.py.rst [hunk 1]
```diff
@@ -69,6 +69,26 @@ To achieve this, we use ensemble parallelism, which involves solving simultaneou
...
+We start  by importing firedrake and the necessary parameters used in spatial and time executions::
+
+    from firedrake import *
+    import os
+    test = False
+    if os.getenv("FIREDRAKE_CI_TESTS") == "1":
+        test = True
+
+    def spatial_time_parameters(test):
+        if test:
```
Comment (dham) [inline demos/full_waveform_inversion/full_waveform_inversion.py.rst:81]:
```suggestion

    def spatial_time_parameters():
        import os
        if os.getenv("FIREDRAKE_CI_TESTS") == "1":
```

File: demos/full_waveform_inversion/full_waveform_inversion.py.rst [hunk 2]
```diff
@@ -99,19 +119,17 @@ The source number is defined with the ``Ensemble.ensemble_comm`` rank::
...
 
 In this example, we consider a two-dimensional square domain with a side length of 1.0 km. The mesh is
-built over the ``my_ensemble.comm`` (spatial) communicator::
-    
-    Lx, Lz = 1.0, 1.0
-    mesh = UnitSquareMesh(80, 80, comm=my_ensemble.comm)
+built over the ``my_ensemble.comm`` (spatial) communicator.::
 
-The basic input for the FWI problem are defined as follows::
+    dt, final_time, nx, ny = spatial_time_parameters(test)
```
Comment (dham) [inline demos/full_waveform_inversion/full_waveform_inversion.py.rst:124]:
```suggestion
    dt, final_time, nx, ny = spatial_time_parameters()
```

**Review by dham — CHANGES_REQUESTED** [review]

File: demos/full_waveform_inversion/full_waveform_inversion.py.rst (hunk 1 above)
Comment (Ig-dolci) [reply demos/full_waveform_inversion/full_waveform_inversion.py.rst]: done.

**Review by connorjward — DISMISSED** [review]
Thanks!

**Review by connorjward — APPROVED** [review]

**Pull request merged**